src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -136,15 +135,16 @@ public int OnActiveProjectCfgChange(IVsHierarchy pIVsHierarchy)
         #region IDisposable
         public void Dispose()
         {
-            var buildManager = _buildManager;
-            if (buildManager != null)
+            // Build manager APIs require UI thread access.
+            ThreadHelper.Generic.Invoke(() =>
```

Comment:
You already have an IProjectThreadingService instance, just use that to switch to the UI thread:

```C#
_threadingService.ExecuteSynchronously(async () => 
{
    await _threadingService.SwitchToUIThread();
    // Dispose
});